Classes/IdentityToken.php

```diff
@@ -190,14 +190,14 @@ private function verifyRsaJwtSignature(string $hashType, array $jwk, string $pay
      *
      * @param array $keys
      * @param string $algorithm
-     * @param string $keyIdentifier
+     * @param string|null $keyIdentifier
      * @return array
      * @throws ServiceException
      */
-    private function getMatchingKeyForJws(array $keys, string $algorithm, string $keyIdentifier): array {
+    private function getMatchingKeyForJws(array $keys, string $algorithm, ?string $keyIdentifier): array {
         foreach ($keys as $key) {
             if ($key['kty'] === 'RSA') {
-                if (!isset($header['kid']) || $key['kid'] === $keyIdentifier) {
+                if ($keyIdentifier !== null || $key['kid'] === $keyIdentifier) {
```

Comment:
Wouldn't this throw an error if `$header['kid']` is not set?

Reply:
dang.. I tested this locally but mixed up the condition in this PR
```suggestion
                if ($keyIdentifier === null || $key['kid'] === $keyIdentifier) {
```